### ISSUE TITLE
Black screen when using screenshot()

### ISSUE THREAD

**Issue by alvinbengeorge** [asweigart/pyautogui#556]
Hi, I have Fedora Linux in my device, and I recently used pyautogui screenshot() function and when I checked the image in my home directory, the png was black entirely.
I installed Scrot for the compatibility issues from rpmsphere repository using dnf.
https://fedora.pkgs.org/33/rpm-sphere-x86_64/scrot-0.8-21.x86_64.rpm.html
![image](https://user-images.githubusercontent.com/69302420/115120257-35d08400-9fca-11eb-99e7-8d3886d085cd.png)
Followed instructions from this website.
Also, locateOnScreen and locateCenterOnScreen doesnt seem to work, It always returns None(I even decreased the confidence level all the way to 0.2 and made grayscale as True).

The program works fine in Windows.
Thanks in advance

**Comment by alvinbengeorge** [comment]
> Hi, I have Fedora Linux in my device, and I recently used pyautogui screenshot() function and when I checked the image in my home directory, the png was black entirely.
> I installed Scrot for the compatibility issues from rpmsphere repository using dnf.
> https://fedora.pkgs.org/33/rpm-sphere-x86_64/scrot-0.8-21.x86_64.rpm.html
> ![image](https://user-images.githubusercontent.com/69302420/115120257-35d08400-9fca-11eb-99e7-8d3886d085cd.png)
> Followed instructions from this website.
> Also, locateOnScreen and locateCenterOnScreen doesnt seem to work, It always returns None(I even decreased the confidence level all the way to 0.2 and made grayscale as True).
> 
> The program works fine in Windows.
> Thanks in advance

I have also found out that its the fault of pyscreeze. Scrot invalid option -z

**Comment by majojoe** [comment]
I have the same problem on a Kubuntu 20.04LTS system (Displaymanager: sddm). I have read anywhere that this might have s.th. to do with wayland, but don't know if that is correct.
Most of the time the screenshot function produces a correct screenshot, but sometimes it produces an entirely black image.
Could you provide a solution, please.

**Comment by ajinzrathod** [comment]
> Hi, I have Fedora Linux in my device, and I recently used pyautogui screenshot() function and when I checked the image in my home directory, the png was black entirely.
> I installed Scrot for the compatibility issues from rpmsphere repository using dnf.
> https://fedora.pkgs.org/33/rpm-sphere-x86_64/scrot-0.8-21.x86_64.rpm.html
> ![image](https://user-images.githubusercontent.com/69302420/115120257-35d08400-9fca-11eb-99e7-8d3886d085cd.png)
> Followed instructions from this website.
> Also, locateOnScreen and locateCenterOnScreen doesnt seem to work, It always returns None(I even decreased the confidence level all the way to 0.2 and made grayscale as True).
> 
> The program works fine in Windows.
> Thanks in advance

Exactly Same Problem. I have tested it in `Ubuntu 18`, `Ubuntu 20`, `Windows-10` and `MacOS Big Sur` and it works perfectly everywhere but Fedora.

BTW, I am using `Fedora 34 5.11.12-300.fc34.x86_64`

**Comment by alvinbengeorge** [comment]
> > Hi, I have Fedora Linux in my device, and I recently used pyautogui screenshot() function and when I checked the image in my home directory, the png was black entirely.
> > I installed Scrot for the compatibility issues from rpmsphere repository using dnf.
> > https://fedora.pkgs.org/33/rpm-sphere-x86_64/scrot-0.8-21.x86_64.rpm.html
> > ![image](https://user-images.githubusercontent.com/69302420/115120257-35d08400-9fca-11eb-99e7-8d3886d085cd.png)
> > Followed instructions from this website.
> > Also, locateOnScreen and locateCenterOnScreen doesnt seem to work, It always returns None(I even decreased the confidence level all the way to 0.2 and made grayscale as True).
> > The program works fine in Windows.
> > Thanks in advance
> 
> Exactly Same Problem. I have tested it in `Ubuntu 18`, `Ubuntu 20`, `Windows-10` and `MacOS Big Sur` and it works perfectly everywhere but Fedora.
> 
> BTW, I am using `Fedora 34 5.11.12-300.fc34.x86_64`

yeah I use the same Fedora, the problem is with scrot, they removed an option in scrot, which i downloaded from dnf, maybe I'll need to install scrot from somewhere else like flatpak
Thank u for replying

**Comment by Avasam** [comment]
Using Ubuntu 22.04 inside VMWare and I'm having the same issue.
